include/nbl/ui/CWindowManagerX11.h

```diff
@@ -69,6 +70,38 @@ static Xinput xinput("Xinput");
 static Xrandr xrandr("Xrandr");
 static Xxf86vm xxf86vm("Xxf86vm");
 
+template<typename Key, typename Val>
+struct MultithreadedMap
+{
+    public:
+        MultithreadedMap() = default;
+        MultithreadedMap(const MultithreadedMap&) = delete;
+        MultithreadedMap(MultithreadedMap&&) = delete;
+        MultithreadedMap& operator=(const MultithreadedMap&) = delete;
+        MultithreadedMap& operator=(MultithreadedMap&&) = delete;
+
+        mutable system::SReadWriteSpinLock m_lock;
+
+        std::map<Key, Val> m_map;
+
+        inline void insert(const Key& _key, const Val& _val)
+        {
+            auto lk = lock_write();
+            m_map.insert(std::make_pair(_key, _val));
+        }
+
+        inline Val& read(const Key& _object)
```

Comment:
call it `get`

Reply:
or `find`

Reply:
use the `m_map.find` method and handle the case where object is not found `found==end()`, if the function MUST always find the object (by logical construction) then `assert`